netket/nn/symmetric_linear.py

```diff
@@ -211,35 +572,216 @@ def full_bias(self, bias):
+        if shape is None:
+            raise TypeError(
+                "When requesting `mode=fft`, the shape of the translation group must be specified. "
+                "Either supply the `shape` keyword argument or pass a `netket.graph.Graph` object to "
+                "the symmetries keyword argument."
+            )
+        else:
+            return DenseSymmFFT(sym, shape=shape, **kwargs)
+    else:
+        return DenseSymmMatrix(sym, **kwargs)
+
+
+def DenseEquivariant(symmetries, mode="auto", shape=None, point_group=None, **kwargs):
+    r"""A group convolution operation that is equivariant over a symmetry group
+
+    Acts on a feature map of symmetry poses of shape [batch_size,n_symm*in_features]
+    and returns a feature  map of poses of shape [batch_size,n_symm*out_features]
+
+    G-convolutions are described in ` Cohen et. {\it al} <http://proceedings.mlr.press/v48/cohenc16.pdf>`_
+    and applied to quantum many-body problems in ` Roth et. {\it al} <https://arxiv.org/pdf/2104.05085.pdf>`_
+
+    The G-convolution generalizes the convolution to non-commuting groups:
+
+    .. math ::
+
+        C^i_g = \sum_h {\bf W}_{g^{-1} h} \cdot {\bf f}_h
+
+    Group elements that differ by the same symmetry operation (i.e. :math:`g = xh`
+    and :math:`g' = xh'`) are connected by the same filter.
+
+    Args:
+        symmetries: A specification of the symmetry group. Can be given by a
+            nk.graph.Graph, an nk.utils.PermuationGroup, a list of irreducible
+            representations or a product table
+        mode: string "fft, irreps, matrix, auto" specifying whether to use a fast
+            fourier transform over the translation group, a fourier transform using
+            the irreducible representations or by constructing the full kernel matrix.
+        features: The number of symmetry-reduced features. The full output size
+            is n_symm*features.
+        use_bias: A bool specifying whether to add a bias to the output (default: True).
+        mask: An optional array of shape [n_sites] consisting of ones and zeros
+            that can be used to give the kernel a particular shape
+        dtype: The datatype of the weights. Defaults to a 64bit float
+        precision: Optional argument specifying numerical precision of the computation
+            see `jax.lax.Precision`for details.
+        kernel_init: Optional kernel initialization function. Defaults to variance scaling
+        bias_init: Optional bias initialization function. Defaults to zero initialization
+    """
 
-        return y
+    if isinstance(symmetries, Graph):
+        # With graph try to find point group, otherwise default to automorphisms
+        if point_group:
+            sg = symmetries.space_group(point_group)
+            if mode == "auto":
+                mode = "fft"
+        elif symmetries._point_group:
+            sg = symmetries.space_group()
+            if mode == "auto":
+                mode = "fft"
```

Comment:
those two can be conflated i think

Reply:
by that i mean just check 
```
if point_group is not None or symmetries._point_group is not None:
```

Also, always check explicitly for None.

Reply:
> Also, always check explicitly for None.

I'll make sure I did this everywhere. 

I'd rather keep it as is, where a user defined point group supersedes the default point group

Reply:
As i said above.
The change I'm proposing does not change the behaviour.
It simply is more coincise.

The logic to handle `point_group = None` or `point_groiup != None` is in `Lattice.spacegroup`...
I don't like it. But you and @attila-i-szabo wanted it there.
So don't duplicate it here.

Reply:
I mean. 
By @attila-i-szabo 's implementation,
```python
symmetries.space_group()
```
is equivalent to
```python
symmetries.space_group(None)
```

So doing
```
if pg is None:
   return symmetries.space_group()
else:
   return symmetries.space_group(pg)
```
is redundant. you check for None twice. 
I'm simply asking to rely on the behaviour that is already implemented instead of reimplementing it.

BTW. If this is not evident to you too, it confirms my suspicion that this logic is hidden/confusing and we'd need something better. But since we don't have better ideas... let's just stick with it for the time being.

Reply:
Ahh, I wasn't aware of that. I'll fix that